ls1intum/ArTEMiS
# feature/participation-websocket-service

…ion for exercise details page and result component, separated result component in ui and logic component




### Checklist

### Motivation and Context
The websocket connection does not work correctly if there are more than one similar connection from the same client at the same time. This pull requests adds a central service that take care of the websocket connections.

### Description
The participation websocket service handles all participation related websocket connections. It then distributes the information to the corresponding components that listens for that change. Now each result component is updated with that service. 

### Steps for Testing
1. Log in to ArTEMiS
2. Create and perform exercises
3. Rate the in a different tab. 
4. the results should appear in all windows
